src/main/java/me/lokka30/levelledmobs/listeners/EntityDeathListener.java

```diff
@@ -60,14 +65,22 @@ public void onDeath(@NotNull final EntityDeathEvent event) {
 
         if(lmEntity.isLevelled() || main.rulesManager.getRule_UseCustomDropsForMob(lmEntity).useDrops){
             //ChunkKey copied from Paper source code
-            int _X=(int)(lmEntity.getLivingEntity().getChunk().getX()>>4),_Z=(int)(lmEntity.getLivingEntity().getChunk().getZ()>>4);
-            long chunkKey=(long) _X & 0xffffffffL | ((long) _Z & 0xffffffffL) << 32;
+            int _X = (int)(lmEntity.getLivingEntity().getChunk().getX()>>4), _Z = (int)(lmEntity.getLivingEntity().getChunk().getZ()>>4);
+            long chunkKey = (long) _X & 0xffffffffL | ((long) _Z & 0xffffffffL) << 32;
             if(!main.entityDeathInChunkCounter.containsKey(chunkKey)){
-                main.entityDeathInChunkCounter.put(chunkKey,new ArrayList<>());
+                main.entityDeathInChunkCounter.put(chunkKey,new IndexedTreeSet<>());
             }
             main.entityDeathInChunkCounter.get(chunkKey).add(new MutablePair<>(new Timestamp(System.currentTimeMillis()),lmEntity));
         }
 
+        int numberOfEntityDeathInChunk=Utils.getNumberOfEntityDeathInChunk(lmEntity,main,main.maximumCoolDownTime);
+        // Only send message for maximum threshold and cool down time
+        if(numberOfEntityDeathInChunk>main.maximumDeathInChunkThreshold) {
+            if(lmEntity.getLivingEntity().getKiller() != null && lmEntity.getLivingEntity().getKiller() instanceof Player)
+                lmEntity.getLivingEntity().getKiller().
+                    sendMessage(MessageUtils.colorizeAll(main.messagesCfg.getString("other.no-drop-in-chunk"))); // Might be spamming
```

Comment:
I'm unsure if we should send a message or not

Reply:
Maybe add an setting?

Reply:
Sounds good, should it be enabled by default?

Reply:
we don't need the comment 'might be spamming' anymore